docs/proposals/batch-node-process.md

```diff
@@ -0,0 +1,213 @@
+
+- `Config.Keadm.Nodes.NodeName`: The name of the remote node. It is not mandatory to be consistent with the actual node name, and it is mainly used to distinguish different node tasks. This parameter must be unique and cannot be repeated.
+- `Config.Keadm.Nodes.KeadmCmd`: The keadm command to be executed. The command does not include the first-level keadm command. The template parameters configured in `Config.Keadm.CmdTplArgs` can be referenced in this command. For specific usage, refer to the following case.
+  **Note**: The wildcard is used as `{{xxx}}`. When the wildcard appears at the beginning of the command, the command needs to be enclosed in `''` or `""`.
+- `Config.Keadm.Nodes.CopyFrom`: The file copy directory, an optional parameter. Once this parameter is configured, all files in this directory will be copied to the remote node. Currently, subdirectories are not allowed in this directory.
+- `Config.Keadm.Nodes.SSH`: Configures the SSH login information of the remote node, including `IP`, `Username`, `Port`, `Auth`, etc. The `Username` needs to be configured as the `root` user to ensure relevant permissions. `Port` is an optional parameter, and the default value is `22`. `Auth` supports two types: `privateKey` and `password`.
+
+`Config.Keadm.MaxRunNum`: Used to configure the maximum number of goroutines when go executes in batches.
+
+#### Overall Execution Flow Chart
+
+![2821732519939_.pic](../images/batch-node-process/batch-node-process-flow-diagram.png)
+
+**Control node**:
+
+- Responsible for downloading the keadm installation package according to the specific configuration or using the offline installation package. The downloaded installation package is uniformly stored in the `/tmp/kubeedge/keadm/package/{amd64/arm64/arm}` directory of the control node.
+- If the user configures the offline installation package path by themselves, they need to create the offline file according to the format requirement of `xxx/{amd64/arm64/arm}/keadm-{version}-linux-{arch}.tar.gz`.
+- The installation package is uniformly decompressed to the `/tmp/kubeedge/keadm/bin/{amd64/arm64/arm}` directory.
+
+**Join node**:
+
+- The node access process is executed concurrently, and the concurrency number depends on the configuration of `Config.Keadm.MaxRunNum`.
+- The keadm executable file and all files in the `Config.Keadm.Nodes.CopyFrom` directory are uploaded to the `/tmp/kubeedge/keadm` directory of the node to be accessed through ssh.
+
+#### Pre-run and Post-run Parameter Description
+
+To support custom scripts, we have introduced the optional parameters `pre-run` and `post-run` in various commands of keadm. These parameters allow users to pass in custom script files to perform some pre or post script tasks. For specific usage, refer to the subsequent use cases.
+
+Usage command reference:
+
+- keadm reset --post-run=test.sh xxx
+- keadm join --pre-run=test.sh xxx
+
+## Use Example
```

Comment:
Could we move this part to our website docs? Putting it here will make the proposal too complicated.

Reply:
Yes, we can also write a promotional blog post for this part

Reply:
We can link to the content here when the subsequent documentation site is ready.

Reply:
Use cases have been linked to the documentation site